File: cpu.py
from perangkat import Perangkat

class CPU(Perangkat):
    def __init__(self, merk, nama, jumlah_core, kecepatan_ghz):
        super().__init__(merk, nama)
        self.jumlah_core = jumlah_core
        self.kecepatan_ghz = kecepatan_ghz
    
    def info(self):
        return f"CPU: {self.merk} {self.nama} ({self.jumlah_core} Core, {self.kecepatan_ghz}GHz)"

File: cpu.cpp
#include "cpu.h"

CPU::CPU(std::string merk, std::string nama, int jumlah_core, double kecepatan_ghz)
    : Perangkat(merk, nama), jumlah_core(jumlah_core), kecepatan_ghz(kecepatan_ghz) {}

std::string CPU::info() const {
    return "CPU: " + merk + " " + nama + " (" + std::to_string(jumlah_core) +
           " Core, " + std::to_string(kecepatan_ghz) + "GHz)";
}
